src/yachtzee/diet.ts

```diff
@@ -0,0 +1,692 @@
+    left.spleen,
+    left.action
+  );
+}
+
+function castOde(turns: number): boolean {
+  if (!have($skill`The Ode to Booze`)) return false;
+
+  shrugIrrelevantSongs();
+
+  // If we have the polka of plenty skill, we can re-buff up later
+  // Else, get rid of chorale which is the most inefficient song
+  if (getActiveSongs.length === (have($skill`Mariachi Memory`) ? 4 : 3)) {
+    if (have($skill`The Polka of Plenty`)) cliExecute(`shrug ${$effect`Polka of Plenty`}`);
+    else cliExecute(`shrug ${$effect`Chorale of Companionship`}`);
+  }
+
+  while (haveEffect($effect`Ode to Booze`) < turns) {
+    useSkill($skill`The Ode to Booze`);
+  }
+  return true;
+}
+
+export function executeNextDietStep(stopBeforeJellies?: boolean): void {
+  if (get("_stenchJellyUsed", false)) return;
+  print("Executing next diet steps", "blue");
+  const dietUtil = new YachtzeeDietUtils();
+  dietUtil.resetDietPref();
+  const VOA = get("valueOfAdventure");
+
+  const dietString = get("_garboYachtzeeChainDiet").split(",");
+  let stenchJellyConsumed = false;
+  for (const name of dietString) {
+    if (name.length === 0) continue;
+    else if (
+      (!stenchJellyConsumed && name.includes("stench jelly")) ||
+      ["clara's bell", "jurassic parka"].includes(name)
+    ) {
+      if (stopBeforeJellies) dietUtil.addToPref(1, name);
+      else {
+        const entry = dietUtil.dietArray.find((entry) => entry.name === name);
+        if (entry) {
+          if (entry.fullness > 0) {
+            if (!get("_milkOfMagnesiumUsed")) {
+              acquire(1, $item`milk of magnesium`, 5 * VOA);
+              use(1, $item`milk of magnesium`);
+            }
+            if (!get("_distentionPillUsed") && have($item`distention pill`)) {
+              use(1, $item`distention pill`);
+            }
+          }
+          entry.action(1);
+        } else {
+          throw new Error(`Could not find ${name} in dietArray`);
+        }
+        set("_stenchJellyUsed", true);
+      }
+      stenchJellyConsumed = true;
+    } else if (!stenchJellyConsumed) {
+      dietUtil.dietArray.forEach((entry) => {
+        if (entry.name === name) {
+          if (myFullness() + entry.fullness > fullnessLimit()) {
+            throw new Error(`consuming ${entry.name} will exceed our fullness limit`);
+          } else if (myInebriety() + entry.drunkenness > inebrietyLimit()) {
+            throw new Error(`consuming ${entry.name} will exceed our inebriety limit`);
+          } else if (mySpleenUse() + entry.spleen > spleenLimit()) {
+            throw new Error(`consuming ${entry.name} will exceed our spleen limit`);
+          }
+          if (entry.fullness > 0) {
+            if (!get("_milkOfMagnesiumUsed")) {
+              acquire(1, $item`milk of magnesium`, 5 * VOA);
+              use(1, $item`milk of magnesium`);
+            }
+            if (!get("_distentionPillUsed") && have($item`distention pill`)) {
+              use(1, $item`distention pill`);
+            }
```

Comment:
Finally, support for sweat outs and synthetic pills can be triggered here once we have sufficient inebriety. This should be an "always on" feature much like the distention pill.